packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -185,6 +187,13 @@ const RenderForm = ({ track }: { track: PurchasableTrackMetadata }) => {
 
   const isPurchaseSuccessful = stage === PurchaseContentStage.FINISH
 
+  // Navigate to track screen in the background if purchase is successful
+  useEffect(() => {
+    if (isPurchaseSuccessful) {
+      navigation.navigate('Track', { id: track.track_id })
```

Comment:
Does this work correctly if you're already on the track page? On web, I had to use a specific helper to avoid creating a duplicate item in the nav stack.

Reply:
oh interesting. will check

Reply:
works correctly still